front/lib/api/assistant/configuration/table_query.ts

```diff
@@ -77,3 +87,90 @@ export async function fetchTableQueryActionConfigurations({
 
   return actionsByConfigurationId;
 }
+
+export async function createTableDataSourceConfiguration(
+  auth: Authenticator,
+  tableConfigurations: TableDataSourceConfiguration[],
+  tablesQueryConfig: AgentTablesQueryConfiguration,
+  t: Transaction
+) {
+  // Although we have the capability to support multiple workspaces,
+  // currently, we only support one workspace, which is the one the user is in.
+  // This allows us to use the current authenticator to fetch resources.
+  const allWorkspaceIds = [
+    ...new Set(tableConfigurations.map((tc) => tc.workspaceId)),
+  ];
+  const hasUniqueAccessibleWorkspace =
+    allWorkspaceIds.length === 1 &&
+    auth.getNonNullableWorkspace().sId === allWorkspaceIds[0];
+
+  assert(
+    hasUniqueAccessibleWorkspace,
+    "Can't create TableDataSourceConfiguration for query tables: Multiple workspaces."
+  );
+
+  const globalVault = await VaultResource.fetchWorkspaceGlobalVault(auth);
+
+  const dataSourceIds = tableConfigurations.map((tc) => tc.dataSourceId);
+  const dataSources = await DataSourceResource.fetchByNames(
+    // We can use `auth` because we limit to one workspace.
+    auth,
+    dataSourceIds
+  );
+
+  const uniqueDataSources = _.uniqBy(dataSources, (ds) => ds.id);
+
+  const dataSourceViews =
+    await DataSourceViewResource.listForDataSourcesInVault(
+      // We can use `auth` because we limit to one workspace.
+      auth,
+      uniqueDataSources,
+      globalVault
+    );
+
+  return Promise.all(
+    tableConfigurations.map(async (tc) => {
+      const dataSource = dataSources.find((ds) => ds.name === tc.dataSourceId);
+      assert(
+        dataSource,
+        "Can't create TableDataSourceConfiguration for query tables: DataSource not found."
+      );
+
+      let dataSourceView;
+
+      // Since the UI does not currently provide the data source view,
+      // we try to retrieve the view associated with the data from the global vault
+      // and assign it to the table data source configuration.
+      if (!tc.dataSourceViewId) {
+        dataSourceView = dataSourceViews.find(
+          (dsv) => dsv.dataSourceId === dataSource.id
+        );
+      } else {
+        dataSourceView = dataSourceViews.find(
+          (dsv) => dsv.sId === tc.dataSourceViewId
+        );
+      }
+
+      assert(
+        dataSourceView,
+        "Can't create TableDataSourceConfiguration for query tables: DataSourceView not found."
+      );
+      assert(
+        dataSourceView.dataSource.sId === tc.dataSourceId,
+        "Can't create TableDataSourceConfiguration for query tables: data source view does not belong to the data source."
+      );
+
+      await AgentTablesQueryConfigurationTable.create(
+        {
+          // TODO(GROUPS_INFRA) Use ModelId for dataSourceId.
+          dataSourceId: dataSource.sId,
```

Comment:
let's be explicit and use dataSource.name to avoid introducing .sId() that points to `name` especially as this will be refactord to dataSource.id next I presume?